src/main.rs

```diff
@@ -1029,33 +1124,45 @@ fn read_events(stream: &mut impl Write, path: &Path, rdesc: &ReportDescriptor) -
     let mut start_time: Option<Instant> = None;
     let mut last_timestamp: Option<Instant> = None;
     let mut data = [0; 1024];
+    let mut ringbuf: Option<libbpf_rs::RingBuffer> = None;
+
+    if let Some(map_ringbuf) = map_ringbuf {
+        let mut builder = libbpf_rs::RingBufferBuilder::new();
+        builder
+            .add(map_ringbuf, move |data| {
+                event_handler(data, &mut start_time)
+            })
+            .unwrap();
+        ringbuf = Some(builder.build().unwrap());
+    }
```

Comment:
```suggestion
    let ringbuf: Option<libbpf_rs::RingBuffer> = if let Some(map_ringbuf) = map_ringbuf {
        let mut builder = libbpf_rs::RingBufferBuilder::new();
        builder
            .add(map_ringbuf, move |data| {
                event_handler(data, &mut start_time)
            })
            .unwrap();
        Some(builder.build().unwrap())
    } else {
        None
    }
```

though I think what should also work is:
```rs
let ringbuf = map_ringbuf.map(|map_ringbuf| {
   let mut builder ...
});
```

Reply:
that last suggestion makes the code much cleaner, thanks!